ngraph/core/reference/include/ngraph/runtime/reference/gather_tree.hpp

```diff
@@ -0,0 +1,118 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <numeric>
+
+#include "ngraph/coordinate_transform.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            // This is a implementation of the algorithm from the tensorflow 1.5 sources.
+            template <typename T>
+            void gather_tree(const T* step_ids,
```

Comment:
Please remove template arguments for this method and use memcpy and element_size to copy data.
As example you can use implementations which already don't use template code: reshape, slice, tile and etc.

Reply:
Done.